mypy/main.py

```diff
@@ -102,120 +104,152 @@ def process_options(args: List[str]) -> Tuple[List[BuildSource], Options]:
+                        " or with incomplete type annotations")
+    parser.add_argument('--implicit-any', action='store_true',
+                        help="behave as though all functions were annotated with Any")
+    parser.add_argument('--fast-parser', action='store_true',
+                        help="enable experimental fast parser")
+    parser.add_argument('-f', '--dirty-stubs', action='store_true',
+                        help="don't warn if typeshed is out of sync")
+    parser.add_argument('--pdb', action='store_true', help="invoke pdb on fatal error")
+    parser.add_argument('--use-python-path', action='store_true',
+                        help="search for modules in sys.path of running Python")
+    parser.add_argument('--stats', action='store_true', help="dump stats")
+    parser.add_argument('--inferstats', action='store_true', help="dump type inference stats")
+    parser.add_argument('--custom-typing', metavar='MODULE', help="use a custom typing module")
+
+    report_group = parser.add_argument_group(
+        title='report generation',
+        description='Generate a report in the specified format.')
+    report_group.add_argument('--html-report', metavar='DIR')
+    report_group.add_argument('--old-html-report', metavar='DIR')
+    report_group.add_argument('--xslt-html-report', metavar='DIR')
+    report_group.add_argument('--xml-report', metavar='DIR')
+    report_group.add_argument('--txt-report', metavar='DIR')
+    report_group.add_argument('--xslt-txt-report', metavar='DIR')
+
+    code_group = parser.add_argument_group(title='How to specify the code to type check')
+    code_group.add_argument('-m', '--module', help="type-check module (may be a dotted name)")
+    code_group.add_argument('-c', '--command', help="type-check program passed in as string")
+    code_group.add_argument('-p', '--package', help="type-check all files in a directory")
+    code_group.add_argument('files', nargs='*', help="type-check given files or directories")
+
+    args = parser.parse_args()
+
+    # Check for invalid argument combinations.
+    code_methods = sum(bool(c) for c in [args.module, args.command, args.package, args.files])
+    if code_methods == 0:
+        parser.error("Missing target module, package, files, or command.")
+    elif code_methods > 1:
+        parser.error("May only specify one of: module, package, files, or command.")
+
+    if args.use_python_path and args.python_version and args.python_version[0] == 2:
+        parser.error('Python version 2 (or --py2) specified, '
+                     'but --use-python-path will search in sys.path of Python 3')
+
+    if args.fast_parser and (args.py2 or
+                             args.python_version and args.python_version[0] == 2):
+        parser.error('The experimental fast parser is only compatible with Python 3, '
+                     'but Python 2 specified.')
+
+    # Set options.
     options = Options()
```

Comment:
It's a shame that we still need the Options class, but I understand it's typed which is harder for the Namespace returned by parse_args().